### PR TITLE
Allow to disable the signature check

### PR DESCRIPTION
**Pull request by schiessle**
This allows you to recover encryption files even if the signature is broken

Steps to test:

1. Enable server side encryption
2. Create a encrypted file
3. changed "encrypted" in the oc_filecache table for the file to a different value
4. Try to open the file

By default you will get a "bad signature error", if you disable the signature check in the config.php with `'encryption_skip_signature_check' => true` you should be able to open the file again.

### CODE REVIEW

**Review by rullzer — APPROVED** [review]
:see_no_evil:

**Review by tobiasKaminsky — APPROVED** [review]
:+1:

**Comment by MorrisJobke** [comment]
> To avoid headache an inversed boolean would be better: encryption_use_signature_check, which is true by default, but I am fine with this

This makes a lot sense 👍

File: apps/encryption/lib/Crypto/Crypt.php [hunk 1]
```diff
@@ -482,9 +482,15 @@ public function symmetricDecryptFileContent($keyFileContents, $passPhrase, $ciph
 	 * @throws GenericEncryptionException
 	 */
 	private function checkSignature($data, $passPhrase, $expectedSignature) {
+		$skipSignatureCheck = $this->config->getSystemValue('encryption_skip_signature_check', false);
+
 		$signature = $this->createSignature($data, $passPhrase);
-		if (!hash_equals($expectedSignature, $signature)) {
+		$hash = hash_equals($expectedSignature, $signature);
```
Comment (tobiasKaminsky) [inline apps/encryption/lib/Crypto/Crypt.php]: better naming, maybe "correctHash"

File: apps/encryption/lib/Crypto/Crypt.php [hunk 2]
```diff
@@ -557,11 +563,13 @@ private function splitMetaData($catFile, $cipher) {
 	 * @throws GenericEncryptionException
 	 */
 	private function hasSignature($catFile, $cipher) {
+		$skipSignatureCheck = $this->config->getSystemValue('encryption_skip_signature_check', false);
```
Comment (MorrisJobke) [inline apps/encryption/lib/Crypto/Crypt.php]: Please change the name here as well.

File: apps/encryption/lib/Crypto/Crypt.php [hunk 3]
```diff
@@ -482,9 +482,15 @@ public function symmetricDecryptFileContent($keyFileContents, $passPhrase, $ciph
 	 * @throws GenericEncryptionException
 	 */
 	private function checkSignature($data, $passPhrase, $expectedSignature) {
+		$enforceSignature = !$this->config->getSystemValue('encryption_skip_signature_check', false);
```
Comment (tobiasKaminsky) [inline apps/encryption/lib/Crypto/Crypt.php]: Maybe also change it in setting -> "'encryption_use_signature_check', true".

**Comment by MorrisJobke** [comment]
@schiessle Why was this merged even if there were open comments and the label was "2. developing"?

**Comment by wasilla** [comment]
Could this be a security issue?

**Pull request merged**